server/rest/submission.py

```diff
@@ -329,6 +329,12 @@ def postSubmission(self, phase, folder, params):
         .errorResponse('Site admin access is required.', 403)
     )
     def updateSubmission(self, submission, params):
+        # Ensure write access on the containing challenge phase, in case this
+        # endpoint is ever opened to non-site-admins
+        user = self.getCurrentUser()
+        phase = self.model('phase', 'covalic').load(
+            submission['phaseId'], user=user, exc=True, level=AccessType.WRITE)
```

Comment:
Sorry, I think  my previous comment got burred by GitHub.

[`Submission`](https://github.com/girder/covalic/blob/master/server/models/submission.py#L29) is just a plain `Model`, not an `AccessControlledModel` or `AccessControlMixin`, so I don't believe I can use the `level` argument to the `loadmodel` decorator for it.

Of course, I think the best solution is to change `Submission` to use `AccessControlMixin` against it's containing `Phase`, but that's way outside of scope here.